infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -162,13 +164,23 @@ public void increment() {
                 .dataStore(serverContext.getDataStore())
                 .build();
 
+        FileSystemAdvisor fsAdvisor = new FileSystemAdvisor();
+
         healingAgent = HealingAgent.builder()
                 .advisor(advisor)
+                .fsAdvisor(fsAdvisor)
                 .localEndpoint(serverContext.getLocalEndpoint())
                 .runtimeSingleton(runtimeSingletonResource)
                 .failureDetectorWorker(failureDetectorWorker)
                 .dataStore(fdDataStore)
                 .build();
+
+        failuresAgent = FailuresAgent.builder()
+                .localEndpoint(serverContext.getLocalEndpoint())
+                .fdDataStore(fdDataStore)
+                .advisor(advisor)
+                .fsAdvisor(fsAdvisor)
+                .build();
```

Comment:
`.runtimeSingleton(runtimeSingletonResource)` is missing here.

Reply:
done